src/main/java/duke/application/Duke.java

```diff
@@ -0,0 +1,243 @@
+                        System.out.println(task);
+                    } catch (NullPointerException e) {
+                        System.out.println("you have failed to mark the task");
+                    }
+                } else if (userInput.startsWith("deadline")) {
+                    try {
+                        Deadline deadline = getDeadline(userInput);
+                        taskList.add(deadline);
+                        System.out.println(taskAddedSuccessfully);
+                        System.out.println(deadline);
+                        printTotalNumberOfTasks(taskList);
+                    } catch (IllegalDeadlineException e) {
+                        System.out.println("OOPS!!! Deadline must have a description and due date.");
+                    }
+                } else if (userInput.startsWith("event")) {
+                    try {
+                        Event event = getEvent(userInput);
+                        taskList.add(event);
+                        System.out.println(taskAddedSuccessfully);
+                        System.out.println(event);
+                        printTotalNumberOfTasks(taskList);
+                    } catch (IllegalEventException e) {
+                        System.out.println("OOPS!!! Event must have a description and date");
+                    }
+                } else if (userInput.startsWith("todo")) {
+                    try {
+                        Todo todo = getTodo(userInput);
+                        taskList.add(todo);
+                        System.out.println(taskAddedSuccessfully);
+                        System.out.println(todo);
+                        printTotalNumberOfTasks(taskList);
+                    } catch (IllegalTodoException e) {
+                        System.out.println("OOPS!!! The description of todo cannot be empty");
+                    }
+                } else if (userInput.startsWith("delete")) {
+                    try {
+                        int indexToDelete = getIndexToDelete(userInput);
+                        Task task = taskList.get(indexToDelete);
+                        taskList.remove(indexToDelete);
+                        System.out.println("Noted. Olivia has removed this task:");
+                        System.out.println(task);
+                        printTotalNumberOfTasks(taskList);
+                    } catch (IllegalDeleteException e) {
+                        System.out.println("To delete a task, please enter in the following format: delete <Task Number>");
+                    } catch (NumberFormatException e) {
+                        System.out.println("You must enter a task number to delete it");
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("The task number you entered does not exist");
+                    }
+                }
+                userInput = sc.nextLine();
+                checkUserInputValidity(userInput);
```

Comment:
Is this check redundant? Considering the above if statements perform the same checks, perhaps you can use an else statement for any unknown command?